lib/user.js

```diff
@@ -892,20 +895,40 @@ module.exports = function (config, userDB, couchAuthDB, mailer, emitter) {
         }
         user = result.rows[0].doc;
         hash = util.URLSafeUUID();
-        user.forgotPassword = {
-          token: hash,
-          issued: Date.now(),
-          expires: Date.now() + tokenLife * 1000
-        };
+        if(config.getItem('security.tokenPasswordResetType') == 'pin') {
+            user.forgotPassword = {
+                token:  util.fiveDigitPin(),
+                issued: Date.now(),
+                expires: Date.now() + (900 * 1000) //15 minutes expiration
```

Comment:
I think it is best we stick with the same `tokenLife` setting. Sometimes email servers are delayed and 15 minutes may not be long enough.